src/core/shape_inference/include/string_tensor_unpack_shape_inference.hpp

```diff
@@ -0,0 +1,58 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <cmath>
+
+#include "openvino/op/string_tensor_unpack.hpp"
+#include "shape_infer_type_utils.hpp"
+#include "utils.hpp"
+
+namespace ov {
+namespace op {
+namespace v15 {
+namespace util {
+static Tensor get_string_tensor(const Node* op, size_t idx, const ITensorAccessor& tensor_accessor) {
+    if (auto t = tensor_accessor(idx)) {
+        return t;
+    } else {
+        const auto& constant = as_type_ptr<opset1::Constant>(op->get_input_node_shared_ptr(idx));
+        NODE_VALIDATION_CHECK(op, constant != nullptr, "Static shape inference lacks constant data on port ", idx);
+        return constant->get_tensor_view();
+    }
+}
+}  // namespace util
+template <class TShape, class TRShape = result_shape_t<TShape>>
+std::vector<TRShape> shape_infer(const StringTensorUnpack* op,
+                                 const std::vector<TShape>& input_shapes,
+                                 const ITensorAccessor& tensor_accessor = make_tensor_accessor()) {
+    NODE_VALIDATION_CHECK(op, input_shapes.size() == 1);
+    const auto& data_shape = input_shapes[0];
+    auto output_shapes = std::vector<TRShape>(3);
+
+    // output 1 and 2: begins and ends
+    output_shapes[0] = data_shape;
+    output_shapes[1] = data_shape;
+
+    // output 3: symbols
+    const auto string_data = util::get_string_tensor(op, 0, tensor_accessor);
+    if (string_data) {
+        uint64_t string_count = 1;
```

Comment:
```suggestion
        uint64_t string_count = string_data->get_size();
```
should provide number of elements in tensor